packages/ra-core/src/store/useStore.ts

```diff
@@ -46,25 +47,62 @@ import { useStoreContext } from './useStoreContext';
  */
 export const useStore = <T = any>(
     key: string,
-    defaultValue?: T
-): useStoreResult<T> => {
+    defaultValue?: T,
+    validate?: ValidateStoreValue<T>
+): UseStoreResult<T> => {
     const { getItem, setItem, subscribe } = useStoreContext();
+    if (
+        typeof validate === 'function' &&
+        typeof defaultValue !== 'undefined' &&
+        !validate(defaultValue)
+    ) {
+        console.warn(`Invalid default value for store key ${key}`);
```

Comment:
I think it should throw in that case. Also, please do that only once (in a useEffect, or in the callback of useState), because otherwise you'll be revalidating the default value at each render, which may be expensive.